src/vaadin-combo-box-dropdown-wrapper.html

```diff
@@ -12,48 +12,51 @@
 
 <dom-module id="vaadin-combo-box-dropdown-wrapper">
   <template>
-    <style>
-      #scroller {
-        overflow: auto;
-
-        /* Fixes item background from getting on top of scrollbars on Safari */
-        transform: translate3d(0, 0, 0);
-
-        /* Enable momentum scrolling on iOS (iron-list v1.2+ no longer does it for us) */
-        -webkit-overflow-scrolling: touch;
-
-        /* Fixes scrollbar disappearing when "Show scroll bars: Always" enabled in Safari */
-        box-shadow: 0 0 0 white;
-      }
-    </style>
-    <vaadin-combo-box-dropdown id="dropdown"
-        hidden="[[_hidden(_items.*, loading)]]"
-        position-target="[[positionTarget]]"
-        on-template-changed="_templateChanged"
-        on-position-changed="_setOverlayHeight"
-        theme="[[theme]]">
-      <template>
-        <div id="scroller" on-click="_stopPropagation">
-          <iron-list id="selector" role="listbox" items="[[_getItems(opened, _items)]]" scroll-target="[[_scroller]]">
-            <template>
-              <vaadin-combo-box-item
-                on-click="_onItemClick"
-                index="[[__requestItemByIndex(item, index)]]"
-                item="[[item]]"
-                label="[[getItemLabel(item)]]"
-                selected="[[_isItemSelected(item, _selectedItem, _itemIdPath)]]"
-                renderer="[[renderer]]"
-                role$="[[_getAriaRole(index)]]"
-                aria-selected$="[[_getAriaSelected(_focusedIndex,index)]]"
-                focused="[[_isItemFocused(_focusedIndex,index)]]"
-                tabindex="-1"
-                theme$="[[theme]]">
-              </vaadin-combo-box-item>
-            </template>
-          </iron-list>
-        </div>
-      </template>
-    </vaadin-combo-box-dropdown>
+    <!-- Stamping the content is postponed until the combo box is opened for the first time -->
+    <template id="dropdown-template" is="dom-if">
+      <vaadin-combo-box-dropdown id="dropdown"
+          hidden="[[_hidden(_items.*, loading)]]"
+          position-target="[[positionTarget]]"
+          on-template-changed="_templateChanged"
+          on-position-changed="_setOverlayHeight"
+          theme="[[theme]]">
+        <template>
+          <style>
```

Comment:
This doesn't seem to work in ShadyDOM, style tag is there in Edge and does not seem to be processed.

Reply:
OK, looks like styles are indeed applied, although style tag is there in markup. Interesting.
For now we can leave it as it is, as long as the tests are passing.